encode.go

```diff
@@ -212,11 +212,68 @@ func (c *locationCache) getLocation(offset int) *time.Location {
 	return location
 }
 
+var infinityTsEnabled = false
+var infinityTsNegative time.Time
+var infinityTsPositive time.Time
+
+const (
+	_infinityString                 = "-infinity"
```

Comment:
I think this should be called something else.  negativeInfinityString, for example.